### PR TITLE
docs: env_preferences

### PR DESCRIPTION
**Pull request by ashu26jha**
Added documentation for `env_preferences`

### CODE REVIEW

File: utils/env_preferences/src/apple.rs [hunk 1]
```diff
@@ -59,6 +59,16 @@ fn get_string(ptr: CFStringRef) -> Result<String, RetrievalError> {
...
 
+/// Retrieves system locales for Apple operating systems, in the order preferred by the
+/// user, it consumes [`CFLocaleCopyPreferredLanguages`](https://developer.apple.com/documentation/corefoundation/1542887-cflocalecopypreferredlanguages)
+/// to copy the languages prefered by the user.
+///
+/// ### Example
+///
+/// ```rust
+/// use env_preferences::get_locales;
+/// let locales: Vec<String> = get_locales().unwrap();
```
Comment (robertbastian) [inline utils/env_preferences/src/apple.rs:70]: this is not a useful example, it doesn't show anything that isn't obvious from the function signature

File: utils/env_preferences/src/linux.rs [hunk 2]
```diff
@@ -49,6 +49,9 @@ impl FromStr for LocaleCategory {
     }
 }
 
+// TODO: Add a function to return all the locales POSIX categories explicitly
+
+/// Retrieves locale for `LC_ALL` POSIX category. Returns other categories unless explicitly set in the thread
```
Comment (robertbastian) [inline utils/env_preferences/src/linux.rs:54]: I don't understand what this means

File: utils/env_preferences/src/apple.rs (hunk 1 above)
Comment (ashu26jha) [reply utils/env_preferences/src/apple.rs:70]: An idea could be to format digits on the basis of locale (default settings)?

File: utils/env_preferences/src/linux.rs (hunk 2 above)
Comment (ashu26jha) [reply utils/env_preferences/src/linux.rs:54]: Actually, when we select `setlocale` which the posix locale type (for eg in this case it is LC_ALL), it only retrieves that locales. 

Here my belief is that users should have option to retrieve all the types (for example LC_ADDRESS etc). The intial thought not to add them was having to call `unsafe` functions, I am much comfortable with `unsafe` as compared to initially starting this project

File: utils/env_preferences/src/apple.rs (hunk 1 above)
Comment (robertbastian) [reply utils/env_preferences/src/apple.rs:70]: just remove it for now

File: utils/env_preferences/src/linux.rs (hunk 2 above)
Comment (robertbastian) [reply utils/env_preferences/src/linux.rs:54]: so does this method return all locale categories?

Comment (ashu26jha) [reply utils/env_preferences/src/linux.rs:54]: will*, I can switch it to return to categories in this PR itself

Comment (robertbastian) [reply utils/env_preferences/src/linux.rs:54]: I'm not suggesting any changes to the code, I just still don't understand the comment. What does "Returns other categories unless explicitly set in the thread" *mean*? Can you rephrase so I can understand?

Comment (ashu26jha) [reply utils/env_preferences/src/linux.rs:54]: Like when I testing this function, I noticed that if locale is set and then retrieved the response it something like:
```rust
// Set the locale
setlocale(LC_ADDRESS, 'de_CH');
// Now retrieving the locale for LC_ALL category
let resp = setlocale(LC_ALL, null)
```

Now, this resp will consist of all the categories. BUT if I do this 

```rust
let resp = setlocale(LC_ALL, null)
```

I will only recieve a single locale.

This has been done to prevent panics

Comment (robertbastian) [reply utils/env_preferences/src/linux.rs:54]: So should it say "Also returns other categories if any are explicitly set in the thread"? I think the use of "unless" is incorrect following your explanation.

**Review by robertbastian — APPROVED** [review]
